In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.mlab as PCA

from sklearn.metrics import roc_curve, auc

import sys
import os


In [11]:
## AUC scores will be imported from thier files. Then the Mean will be looked at for each data combination
direc = "created_data/AUC_Scores"

files = {}
for file in os.listdir(direc):
    name = file[:-8] ## remove the _auc.csv part of the filename
    
    files[name] = pd.read_csv(direc + '/' + file, index_col=0)

In [37]:
means = {}
meanDF = pd.DataFrame()

for f in files:
    des = files[f].describe()
    meanDF = meanDF.append(pd.Series.to_frame(des.loc[des.index[1]]).rename(columns={'mean':f}).T)

 

In [40]:
meanDF

KNN        NB        DT       SVM       MLP
UC        0.787126  0.788256  0.687132  0.685678  0.543941
UCUD      0.916985  0.846848  0.803919  0.542973  0.854395
UCUN      0.916985  0.846848  0.803919  0.543306  0.854395
UCUNUD    0.903497  0.668938  0.531968  0.500052  0.838028
UCUNUDUS  0.903498  0.671222  0.517930  0.504764  0.528577
UCUNUS    0.916985  0.849979  0.812515  0.560393  0.534393
UCUS      0.787133  0.799829  0.700490  0.694953  0.526531
UCUSUD    0.739939  0.647087  0.501841  0.546939  0.633512
UD        0.787126  0.788256  0.687132  0.685678  0.543941
UN        0.787126  0.788256  0.687132  0.685678  0.543941
UNUD      0.828310  0.620864  0.510829  0.508925  0.689310
UNUDUS    0.929381  0.937404  0.812144  0.887406  0.909390
US        0.731045  0.640296  0.652446  0.692030  0.756698
USUD      0.916985  0.846848  0.803919  0.542973  0.854395
USUN      0.895167  0.835919  0.787111  0.811659  0.500000

The highest Mean is the NB on the User Network, Demographic and Sentiment feature set. It performed with an average of 0.93 for AUC. If we look at comparing the data from Lee et. al and their feature set containing User Network and Demographic they have an AUC score of 0.995. This is clearly better than mine with the addition of sentiment. However, comparing mine of UN + UD, the addition of US bumps this from 0.62 to the high of 0.937

In [41]:
meanDF.describe()

KNN         NB         DT        SVM        MLP
count  15.000000  15.000000  15.000000  15.000000  15.000000
mean    0.849819   0.771790   0.686695   0.626227   0.674097
std     0.072977   0.097273   0.120048   0.118938   0.154008
min     0.731045   0.620864   0.501841   0.500052   0.500000
25%     0.787126   0.670080   0.592207   0.542973   0.539167
50%     0.895167   0.788256   0.687132   0.560393   0.633512
75%     0.916985   0.846848   0.803919   0.688854   0.846212
max     0.929381   0.937404   0.812515   0.887406   0.909390